src/Persistence/mapping.py

```diff
@@ -35,55 +33,34 @@ def __new__(cls, *args, **kw):
     __subclasscheck__ = ExtensionClass.__subclasscheck__
 
 
-if six.PY2:  # pragma: no cover
-    # Neither six.with_metaclass nor six.add_metaclass work under
-    # both Python 2 and 3 for us, so we provide to code paths.
-
-    class PersistentMapping(Persistent, _BasePersistentMapping):
-        """Legacy persistent mapping class
-
-        This class mixes in :class:`ExtensionClass.Base` if it is present.
-
-        Unless you actually want ExtensionClass semantics, use
-        :class:`persistent.mapping.PersistentMapping` instead.
-        """
-        __metaclass__ = _Meta
-
-        def __setstate__(self, state):
-            if 'data' not in state:
-                state['data'] = state['_container']
-                del state['_container']
-            self.__dict__.update(state)
-
-else:  # pragma: PY3
+def with_metaclass(meta, *bases):
```

Comment:
Could this not be simplified?
The former complexity came from Python 2 and 3 using different mechanisms to specify a metaclass.

Reply:
I looked at the code and it seemed to complex for an easy refactoring. Any help is welcome.

Reply:
I push a simplification trial: `type()` and `._bases__` give the same result.

Reply:
Thank you!